### PR TITLE
add lekp 2.1 migration

### PR DESCRIPTION
**Pull request by wolfderechter**
## ID
DGS-169

## Description

The subsidy LEKP 2.1 is updated with the following:
 - `indienen pact` has a new deadline of 05 december 2024 23:59 and so is available to create again
 - The period of the subsidy is changed to show `2023 - 2024` with `N.V.T` underneath
 - the `opvolgmoment` is closed with enddate 31/12/2023
 - The subsidy is now available for all 300 steden en gemeenten

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other

## How to test

Visit subsidy module and verify the changes are made to `indienen pact`, `Periode` and `opvolgmoment`.

### CODE REVIEW

File: config/migrations/2024/subsidies/20240315100754-update-lekp-2-1-deadlines.sparql [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+  GRAPH ?g {
+    VALUES ?s {
+      <http://data.lblod.info/id/subsidy-measure-offer-series/a04b604a-86c6-49c0-8b70-1910e3fdfa4c> # LEKP 2.1 series
+    }
+    ?s dct:title ?title .
+  }
+};
+
+# Remove from specific graphs
+INSERT DATA {
```
Comment (Riadabd) [inline config/migrations/2024/subsidies/20240315100754-update-lekp-2-1-deadlines.sparql:34]: Should this be `DELETE DATA` instead?

File: config/migrations/2024/subsidies/20240315100754-update-lekp-2-1-deadlines.sparql [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+    <http://data.lblod.info/id/bestuurseenheden/f8ad83ae8d0d96983b5bddb3c2cb768323b4f5103d95c111f6143c9109034aab> # Zwalm
+    <http://data.lblod.info/id/bestuurseenheden/7269d3607a761c9ce0e226e0b7ea5687324230b65a568d48c81a7acba53998fd> # Zwevegem
+    <http://data.lblod.info/id/bestuurseenheden/dac6ed961e62f29e2706c8213c6c5ee28d07bb719e564bcd47157499561e613b> # Zonhoven
+ }
+
+ ?bestuurseenheid mu:uuid ?uuid .
+ BIND(IRI(CONCAT("http://mu.semte.ch/graphs/organizations/", STR(?uuid), "/LoketLB-subsidies")) as ?newGraph)
+}
+
+# Add to public, to be accesible from all graphs
```
Comment (Riadabd) [inline config/migrations/2024/subsidies/20240315100754-update-lekp-2-1-deadlines.sparql:365]: Missing comma:
```suggestion
}

;

# Add to public, to be accesible from all graphs
```

File: config/migrations/2024/subsidies/20240315100754-update-lekp-2-1-deadlines.sparql [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+WHERE {
+  GRAPH ?g {
+    VALUES ?s {
+      <http://data.lblod.info/id/subsidy-measure-offer-series/a04b604a-86c6-49c0-8b70-1910e3fdfa4c> # LEKP 2.1 series
+    }
+    ?s dct:title ?title .
+  }
+};
+
+# Remove from specific graphs
```
Comment (Riadabd) [inline config/migrations/2024/subsidies/20240315100754-update-lekp-2-1-deadlines.sparql:33]:
```suggestion
}

;

# Remove from specific graphs
```

File: config/migrations/2024/subsidies/20240315100754-update-lekp-2-1-deadlines.sparql [hunk 4]
```diff
@@ -0,0 +1,452 @@
...
+      skos:prefLabel            """Lokaal Energie- en Klimaatpact 2.1""" ;
+      skos:related              <https://www.vvsg.be/kennisitem/vvsg/lokaal-energie-en-klimaatpact> ;
+      m8g:hasCriterion          <http://data.lblod.info/id/criterions/f5ea0615-0b0a-47db-a38c-6097ff80815d> ; # Available only for "gemeenten"
+      cpsv:follows              <http://data.lblod.info/id/subsidy-procedural-steps/a9a20ff7-a727-4b92-8d16-dac36b0c27cd> ;
+      cpsv:follows              <http://data.lblod.info/id/subsidy-procedural-steps/be1168a8-5131-4adb-a913-a4147bd8c6f5> ;
+      lblodSubsidie:heeftReeks  <http://data.lblod.info/id/subsidy-measure-offer-series/a04b604a-86c6-49c0-8b70-1910e3fdfa4c> .
+ }
+};
+
+# Update LEKP 2.1 deadline: remove subsidy period -> 'N.V.T'
```
Comment (Riadabd) [inline config/migrations/2024/subsidies/20240315100754-update-lekp-2-1-deadlines.sparql:393]:
```suggestion
}

;

# Update LEKP 2.1 deadline: remove subsidy period -> 'N.V.T'
```

File: config/migrations/2024/subsidies/20240315100754-update-lekp-2-1-deadlines.sparql [hunk 5]
```diff
@@ -0,0 +1,452 @@
...
+    VALUES ?s {
+      <http://data.lblod.info/id/periodes/46e05ff6-08f4-4ff9-a831-8897de75f25c> # LEKP 2.1 subsidy deadline
+    }
+    ?s m8g:startTime ?startTimeSubsidy .
+    ?s m8g:endTime ?endTimeSubsidy .
+  }
+};
+
+
+# Update LEKP 2.1, Indienen Pact deadline
```
Comment (Riadabd) [inline config/migrations/2024/subsidies/20240315100754-update-lekp-2-1-deadlines.sparql:411]:
```suggestion
}

;

# Update LEKP 2.1, Indienen Pact deadline
```

File: config/migrations/2024/subsidies/20240315100754-update-lekp-2-1-deadlines.sparql [hunk 6]
```diff
@@ -0,0 +1,452 @@
...
+  GRAPH ?g {
+    VALUES ?s {
+      <http://data.lblod.info/id/periodes/33f16980-1d9b-49b9-b26c-d595ecb8689f> # LEKP 2.1 Indienen Pact step deadline
+    }
+    ?s m8g:endTime ?endTimeStep .
+  }
+};
+
+
+# Update LEKP 2.1, opvolgmoment deadline
```
Comment (Riadabd) [inline config/migrations/2024/subsidies/20240315100754-update-lekp-2-1-deadlines.sparql:433]:
```suggestion
}

;

# Update LEKP 2.1, opvolgmoment deadline
```

File: config/migrations/2024/subsidies/20240315100754-update-lekp-2-1-deadlines.sparql [hunk 7]
```diff
@@ -0,0 +1,452 @@
...
+  }
+}
+WHERE {
+  GRAPH ?g {
+    VALUES ?s {
+      <http://data.lblod.info/id/periodes/f8e73b21-7231-494c-93b6-8f58bf69b0bd> # LEKP 2.1 opvolgmoment step deadline
+    }
+    ?s m8g:endTime ?endTimeStep .
+  }
+};
```
Comment (Riadabd) [inline config/migrations/2024/subsidies/20240315100754-update-lekp-2-1-deadlines.sparql:452]: Unneeded comma at the end:
```suggestion
}
```

File: config/migrations/2024/subsidies/20240315100754-update-lekp-2-1-deadlines.sparql (hunk 1 above)
Comment (wolfderechter) [reply config/migrations/2024/subsidies/20240315100754-update-lekp-2-1-deadlines.sparql:35]: Ah no my comment just ended up in the wrong place

File: config/migrations/2024/subsidies/20240315100754-update-lekp-2-1-deadlines.sparql [hunk 8]
```diff
@@ -0,0 +1,382 @@
...
+  }
+}
+WHERE {
+  GRAPH ?g {
+    VALUES ?s {
+      <http://data.lblod.info/id/subsidy-measure-offer-series/a04b604a-86c6-49c0-8b70-1910e3fdfa4c> # LEKP 2.1 series
+    }
+    ?s dct:title ?title .
+  }
+}
```
Comment (Riadabd) [inline config/migrations/2024/subsidies/20240315100754-update-lekp-2-1-deadlines.sparql:31]: The title change here can be reduced to [this line](https://github.com/lblod/app-digitaal-loket/pull/546/files#diff-ab3d226211e3532b5ffda2cdbb1649e82d29f4e8d44b9def4d649d128cbaffbaR298) (where you re-insert the subsidy in the public graph); I don't think it's necessary to have it at the top.

**Review by Riadabd — APPROVED** [review]
Good to go!

**Comment by Riadabd** [comment]
Closing since change was incorporated in https://github.com/lblod/app-digitaal-loket/pull/547.

**Pull request closed without merging**